src/framework/standard/help_commands.rs

```diff
@@ -545,7 +550,11 @@ fn nested_group_command_search<'rec, 'a: 'rec>(
                 },
             }
 
-            group.options.prefixes.iter().any(|prefix| trim_prefixless_group(prefix, name));
+            if !group.options.prefixes.is_empty()
+                && !group.options.prefixes.iter().any(|prefix| trim_prefixless_group(prefix, name))
+            {
```

Comment:
This can be simplified to calling just `any`. If the iterator is empty, `any` will return `false`.

```suggestion
            if !group.options.prefixes.iter().any(|prefix| trim_prefixless_group(prefix, name)) {
```

Reply:
Yes, but that's the problem. This will cause an empty `prefixes` to evaluate the condition to `true` and trigger `continue`, skipping any further recursion on it.

A help execution such as `~help upper su` or `~help upper sub` will break fully then. They have no prefix and thus are not considered and skipped.

`any` on an empty `prefixes` evaluates to `false`, negated by `!` this turns into `true`, the condition is met, `continue` reached.

!group.options.prefixes.is_empty()` will return `true` and negated it will turn into `false`, the condition is `&&` and fails, `continue` is unreached.

Reply:
Ah, I see. I went a little too far with pursuit of optimisation that I didn't consider the change in behaviour if you applied my suggestion. You can disregard my comment.

Reply:
Thanks for paying attention to the code though, I think it's a good way to ensure code is valid :)